feat: added support for spacecraft thrusters

Pedro-Roque opened the pull request:
# 🎉 New feature

## Summary
Adds support for Spacecraft thrusters controlled with a duty cycle signal (such as PWM). Method ported from PX4 Gazebo-Classic SITL implemented by @Jaeyoung-Lim

## Test it
<!--Explain how reviewers can test this new feature manually.-->

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [X] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+//////////////////////////////////////////////////
+SpacecraftThrusterModel::SpacecraftThrusterModel()
+  : dataPtr(std::make_unique<SpacecraftThrusterModelPrivate>())
```
include `<memory>`

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    gzerr << "SpacecraftThrusterModel plugin should be attached to a model "
+           << "entity. Failed to initialize." << std::endl;
```
```suggestion
    gzerr << "SpacecraftThrusterModel plugin should be attached to a model "
          << "entity. Failed to initialize." << std::endl;
```

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    gzwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
```
include `<chrono>`

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+        << msg->velocity_size() << std::endl;
+      return;
+    }
+  }else{
+    return;
```
```suggestion
  }
  else
  {
    return;
```

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+  // Process input
+  double ref_duty_cycle_ = msg->normalized(this->actuatorNumber);
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Ref duty cycle: " << ref_duty_cycle_ << std::endl;
```
lines 100 character maximum

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+  // Calculate cycle start time
+  if (this->samplingTime >= 1.0/this->dutyCycleFrequency) {
+    if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Cycle completed. Resetting cycle start time." << std::endl;
```
lines 100 character maximum

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+  // Calculate sampling time instant within the cycle
+  this->samplingTime = this->simTime - this->cycleStartTime;
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": PWM Period: " << 1.0/this->dutyCycleFrequency << " Cycle Start time: " << this->cycleStartTime << " Sampling time: " << this->samplingTime << std::endl;
```
lines 100 character maximum

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+
+  double ref_duty_cycle = ref_duty_cycle_ * (1.0 / this->dutyCycleFrequency);
+  double force = this->samplingTime <= ref_duty_cycle ? this->maxThrust : 0.0;
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Force: " << force << "  Sampling time: " << this->samplingTime << "  Ref duty cycle: " << ref_duty_cycle_ << std::endl;
```
lines 100 character maximum

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+  // Apply a force to the link.
+  link.AddWorldForce(_ecm,
+                      worldPose->Rot().RotateVector(math::Vector3d(0, 0, force)));
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Value: " << msg->normalized(this->actuatorNumber) << 
```
lines 100 character maximum

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,396 @@
+GZ_ADD_PLUGIN_ALIAS(SpacecraftThrusterModel,
+                          "gz::sim::systems::SpacecraftThrusterModel")
+
+// TODO(CH3): Deprecated, remove on version 8
+GZ_ADD_PLUGIN_ALIAS(SpacecraftThrusterModel,
+                          "ignition::gazebo::systems::SpacecraftThrusterModel")
```
you are targeting `gz-sim8`, you should remove this

Pedro-Roque replied:
@ahcorde actually our setup is using GZ Garden, so maybe I need to keep this for now. Unless it is backwards compatible without it?

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.hh:
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
```
```suggestion
 * Copyright (C) 2024 Open Source Robotics Foundation
```

ahcorde commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.hh:
```diff
@@ -0,0 +1,66 @@
+
+    // Documentation inherited
+    public: void Configure(const Entity &_entity,
+                           const std::shared_ptr<const sdf::Element> &_sdf,
```
include `<memory>`

Pedro-Roque commented:
Thanks @ahcorde for the inputs! I had not yet started cleaning the branch as I had just finished the feature. I will address this in the coming days. Thanks!

Pedro-Roque commented:
@ahcorde should be good for a review check. Let me know what else I should improve on it!

ahcorde commented on tutorials/spacecraft_thrusters.md:
```diff
@@ -0,0 +1,83 @@
+For instructions on how to run the spacecraft model, please refer to the [PX4-Space-Systems](https://github.com/DISCOWER/PX4-Space-Systems) page.
+
+Below is a picture of the simulator:
+![Spacecraft simulator](https://raw.githubusercontent.com/DISCOWER/gz-sim/pr-spacecraft-thrusters/tutorials/files/spacecraft/kth_spacecraft_simulator.png)
```
```suggestion
![Spacecraft simulator](./files/spacecraft/kth_spacecraft_simulator.png)
```

ahcorde commented on tutorials/spacecraft_thrusters.md:
```diff
@@ -0,0 +1,83 @@
+
+## Testing an implementation of a Spacecraft model
+
+Examples of spacecraft models with thrusters were implemented as part of the PX4-Autopilot SITL simulation.
+The spacecraft model can be found in the `PX4-gazebo-models` repository, in the `models/spacecraft_2d/model.sdf` directory.
+
+For instructions on how to run the spacecraft model, please refer to the [PX4-Space-Systems](https://github.com/DISCOWER/PX4-Space-Systems) page.
```
@azeey what do you think about these comments?

Maybe it's also worth it to include a small example here ?

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.hh:
```diff
@@ -0,0 +1,66 @@
+  class SpacecraftThrusterModelPrivate;
+
+  /// \brief This system applies a thrust force to models with spinning
+  /// propellers. See examples/worlds/quadcopter.sdf for a demonstration.
```
I didn't know spacecrafts had propellers :upside_down_face: :rofl: 

On a more serious note its worth pointing users to a citation that describes the dynamics of the plugin.
Also, we usually describe the parameters on most plugins in this section.

See: https://github.com/gazebosim/gz-sim/blob/9e60fdc5dea81344d6b3c9c715ef27f1fc21968a/src/systems/hydrodynamics/Hydrodynamics.hh#L49
for an example.

Also an example would help greatly. I know you've written a tutorial

Pedro-Roque replied:
I may or may not have templated from MulticopterMotorModel :eyes:  

Thanks for the insightful comments, I'll patch it up!

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+  link.AddWorldForce(_ecm,
+    worldPose->Rot().RotateVector(math::Vector3d(0, 0, force)));
+  if (DEBUG && this->actuatorNumber == 0)
+    std::cout << this->actuatorNumber
```
Use `gzdbg`

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+#include "gz/sim/Model.hh"
+#include "gz/sim/Util.hh"
+
+#define DEBUG 0
```
If you use `gzdbg` one can directly control the logging levels based on the verbosity set when running. `gz sim -v 4`  will show the debug output.

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+class gz::sim::systems::SpacecraftThrusterModelPrivate
+{
+  /// \brief Callback for actuator commands.
+  public: void OnActuatorMsg(const msgs::Actuators &_msg);
```
Its worth pointing out that perhaps we should use the ECM for some of this. It was a mistake made by myself to implement the thruster using messages. If we had used the ECM a lot of our tests would be much more deterministic.

Pedro-Roque replied:
In our case we are using these messages from PX4 GZ Bridge, so I'll have to keep it, or otherwise I'd have to re-write quite a bit of code on the other stack. We can keep it like this for now, perhaps? The module should still accept inputs from components - check the top of UpdateForcesAndMoments.

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+  }
+
+  // If the joint or links haven't been identified yet, look for them
+  if (this->dataPtr->jointEntity == kNullEntity)
```
I think you can do these checks in the `Configure` stage.

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+  if (this->dataPtr->jointEntity == kNullEntity ||
+      this->dataPtr->linkEntity == kNullEntity ||
+      this->dataPtr->parentLinkEntity == kNullEntity)
+    return;
```
Log an error here.

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc:
```diff
@@ -0,0 +1,431 @@
+  // skip UpdateForcesAndMoments if needed components are missing
+  bool doUpdateForcesAndMoments = true;
+
+  if (!_ecm.Component<components::WorldPose>(this->dataPtr->linkEntity))
```
I think you can do these checks in the Configure stage as well.

Pedro-Roque replied:
I moved the suggested checks to Configure, expecting that these components are not updated on runtime.

arjo129 reviewed (changes requested):
Thanks for this really awesome contribution. I've marked out some things that need changing. It would be nice if you had an example SDF and some unit tests also. These are useful as they allow us to make sure we don't introduce regressions when we make changes to the core simulation platform.

Pedro-Roque commented:
> Thanks for this really awesome contribution. I've marked out some things that need changing. It would be nice if you had an example SDF and some unit tests also. These are useful as they allow us to make sure we don't introduce regressions when we make changes to the core simulation platform.

Thanks @arjo129 ! Unfortunately right now I'm a bit swamped, so might take a bit more time before I add an example SDF and unit tests. Would this be a hard requirement for now, or is it ok if we can merge this in and then I add them?

On another note: I need this plugin to be available on gz-sim7 (since we are using gazebo garden for now). Should I do a PR targetting that branch too?

arjo129 commented:
I would say unit tests can go in later but we should have an example. At the very least thatll allow me to test it.

Pedro-Roque commented:
@arjo129 I Just added a spacecraft world with the DART spacecraft (mass and inertia were _severely_ reduced for faster visualization. Also updated the tutorial.

Moreover, I think that the mesh file should not be there but hosted on fuel, but I couldn't find my way to upload it there. Do I need extra permissions?

arjo129 commented:
You do need to sign in to upload files to fuel. Tbh you can just use a box if you're strapped for time.

Pedro-Roque commented:
> You do need to sign in to upload files to fuel. Tbh you can just use a box if you're strapped for time.

I'll try to do it now so that it's done properly at once. Standby,

arjo129 commented:
I wont be able to test this till tomorrow. Ipl let you know how to fix it.

Pedro-Roque commented:
> I wont be able to test this till tomorrow. Ill let you know how to fix it.

No worries, let me know and I'll fix it over the weekend.

Pedro-Roque commented:
Thanks @arjo129 , just fixed the above items!

Best,
Pedro

arjo129 commented:
> Also, to target gz garden, should I make a PR to that branch?

We try to avoid adding new features to older versions. If it really is needed we do do backports from time to time. Lets first merge it then it can be cherry picked later.

Pedro-Roque commented:
Sounds good @arjo129 !

Let me know if I should do anything else for this to be merged. Thanks!

arjo129 commented on src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.hh:
```diff
@@ -0,0 +1,66 @@
+  // Forward declaration
+  class SpacecraftThrusterModelPrivate;
+
+  /// \brief This system applies a thrust force to models with RCS-like
```
I may have mentioned it before but document the SDF parameters here and topics that control this plugin. Also point to some citation explaining where this model comes from.

arjo129 reviewed (changes requested):
Its more or less looking good to me just handle the two changes I mentioned.

arjo129 commented:
Your DCO needs fixing also are you basing your work on `gz-sim7` so you should target that branch else the abi checker wont be happy.

Pedro-Roque commented:
> Your DCO needs fixing also are you basing your work on `gz-sim7` so you should target that branch else the abi checker wont be happy.

Yes you are right, thanks for changing the target branch to gz-sim7! So later we cherry-pick this for main?

arjo129 commented:
Yes. We will forward port later. First address the open issue a hand.

https://github.com/gazebosim/gz-sim/pull/2431/checks?check_run_id=26016983492

azeey commented on tutorials/spacecraft_thrusters.md:
````diff
@@ -0,0 +1,118 @@
+Here follows an example instance of the SpacecraftThrusterModel plugin in an SDF file:
+```xml
+    <plugin filename="gz-sim-spacecraft_thruster_model-system" name="gz::sim::systems::SpacecraftThrusterModel">
+      <linkName>thruster_0</linkName>
````
We've been trying to improve consistency on the style we use for SDF parameters. In general,  we prefer `snake_case` unless there's a specific reason to use `camelCase`.

cc @bperseghetti

Pedro-Roque replied:
Hi @bperseghetti , thanks for the PR ! I'll give it a look and do the required changes to the fuel model.

Regarding adding the 2D Model, I probably can also do that and add it to fuel as well, since I already have that model implemented anyway.

Regarding the test - I'll probably need to do those later since I really have no time for it now (other than maybe doing a very basic test). Let me fix the items above and then get to this.

Thanks for all the support!

bperseghetti commented:
@Pedro-Roque you might want to look at this PR https://github.com/DISCOWER/gz-sim/pull/1 I submitted to your branch, it aligns more on the standard names/use of actuators, topics, and sub_topics. Note that the MulticopterMotorModel is getting refactored soon (thanks for reminder and catch on that @azeey) to make it camel_case and also follow all the other systems (so making it how other ones like Ackermann, Joint, JointPosition, etc are, is where this request is coming from).

I also cleaned up some repeated areas of the tutorial and made it more in line with the other existing ones.

In my opinion after making the suggested changes in the PR https://github.com/DISCOWER/gz-sim/pull/1 the only remaining items are:

- Update the model revision number in `examples/worlds/spacecraft.sdf` after uploading the revised dart model to app.gazebosim with the model.sdf using the plugin with updated camel case.

- Consider also uploading the 2D testbench model to app.gazebosim and linking that instead in the tutorial so it doesn't allow for any broken links.  

- Add a test to keep system from possibly breaking (can follow for instance follow how [these tests](https://github.com/gazebosim/gz-sim/blob/gz-sim8/test/integration/ackermann_steering_system.cc) are done)

Pedro-Roque commented:
@bperseghetti @azeey I've now updated everything with respect to the provided comments. Only thing that I'll have to do later is the unit tests. I'd be happy if this could be merged meanwhile, since I have another PR depending on this.

Pedro-Roque commented:
@arjo129 are we ready to merge this in?

bperseghetti commented:
Hey @Pedro-Roque  I would suggest still adding a basic test to get some level of code coverage started. Also I'm assuming your dependent PR is the one in PX4, I can help merge that PX4 PR in for you but first a release of garden would need to be scheduled after this current PR gets in and that can take some time since it's more than just getting this PR merged.

With that being said I would suggest the fastest way to get this current PR merged in and start the "release timeline clock" is to create that super basic test based off what I shared for Ackermann (or really any other test in tree as well) with the goal of testing the functionality and also as many lines of code of this PR as possible (for code coverage).

Pedro-Roque commented:
@bperseghetti I created a test and wanted to run it on my machine. However after following the instructions regarding code coverage, colcon only finds 7 tests in the whole workspace and no tests are found on the gz-sim7 package. 

Is there something obvious I'm missing? I'll share here a log of the commands

arjo129 commented:
How are you building your workspace? Generally I would run `colcon build`. Some times you may need to force cmake to reconfigure itself. Running `colcon test` will take very long. I suggest going inside the `build/gz-sim7/bin` directory. You should see the integration test binaries. You can run your test individually from there.

bperseghetti commented:
> @bperseghetti I created a test and wanted to run it on my machine. However after following the instructions regarding code coverage, colcon only finds 7 tests in the whole workspace and no tests are found on the gz-sim7 package. 
> 
> Is there something obvious I'm missing? I'll share here a log of the commands

Want to just push up to this PR your test that you wrote and we can look at/suggest changes from there?

Pedro-Roque commented:
> How are you building your workspace? Generally I would run `colcon build`. Some times you may need to force cmake to reconfigure itself. Running `colcon test` will take very long. I suggest going inside the `build/gz-sim7/bin` directory. You should see the integration test binaries. You can run your test individually from there.

This is what is populated in the folder:

```bash
╭─  ~/gz_garden_ws/build/gz-sim7/bin ···························  07:38:29 
╰─❯ pwd              
/home/roque/gz_garden_ws/build/gz-sim7/bin

╭─  ~/gz_garden_ws/build/gz-sim7/bin ···························  07:38:30 
╰─❯ ls
runGui
```

@bperseghetti I'd like to have a working test before I push it, but I'd also like to try it out locally first.

arjo129 commented:
Seems like you are not building tests. Please use:
```
colcon build --merge-install
```
NOT:
```
colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
```

Pedro-Roque commented:
> Seems like you are not building tests. Please use:
> 
> ```
> colcon build --merge-install
> ```
> 
> NOT:
> 
> ```
> colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
> ```
> 
> Consider deleteing `build` and `install` folders in your workspace for clean builds.

Hi @arjo129 the command I was running is
```bash
colcon build --cmake-args -DCMAKE_BUILD_TYPE=Coverage --merge-install --packages-select gz-sim7
```

I'll do a clean build and come back to you.

arjo129 commented:
Remove the coverage do a vanilla build.

Pedro-Roque commented:
> Remove the coverage do a vanilla build.

Ok, the vanilla build seems to be building the tests. Maybe this is something we should update the docs with. I'll fix the test and push changes later in the week.

Pedro-Roque commented:
@arjo129 @bperseghetti I've added a unit test now that checks for motion of the model upon setting a non-zero input. The test works as expected.

Let me know if there is anything else needed before merging this in.

arjo129 commented on test/integration/spacecraft.cc:
```diff
@@ -0,0 +1,121 @@
+/////////////////////////////////////////////////
+// Test that commanded motor speed is applied
+// See https://github.com/gazebosim/gz-sim/issues/1175
+TEST_F(SpacecraftTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(InputTest))
```
Do we know if this is necessary?

Pedro-Roque replied:
I've kept it as in the ackermann tests, since these are similar. I can try to remove it to check if it fails on Windows.

Pedro-Roque replied:
@arjo129 test fails on Windows - seems like the vehicle did not move at all there. Thoughts?

arjo129 reviewed (approved):
LGTM. I think @bperseghetti, @azeey and @ahcorde should also take a look.

arjo129 commented:
Yeah just revert and ignore windows for now. I'm not sure about it.

On Thu, Jul 18, 2024, 9:03 PM Pedro Roque ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In test/integration/spacecraft.cc
> <https://github.com/gazebosim/gz-sim/pull/2431#discussion_r1682800396>:
>
> > +    serverConfig.SetSdfFile(sdfFile);
> +
> +    auto server = std::make_unique<Server>(serverConfig);
> +    EXPECT_FALSE(server->Running());
> +    EXPECT_FALSE(*server->Running(0));
> +
> +    using namespace std::chrono_literals;
> +    server->SetUpdatePeriod(1ns);
> +    return server;
> +  }
> +};
> +
> +/////////////////////////////////////////////////
> +// Test that commanded motor speed is applied
> +// See https://github.com/gazebosim/gz-sim/issues/1175
> +TEST_F(SpacecraftTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(InputTest))
>
> @arjo129 <https://github.com/arjo129> test fails on Windows - seems like
> the vehicle did not move at all there. Thoughts?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gazebosim/gz-sim/pull/2431#discussion_r1682800396>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEEMQC3JXL4SF55RMVDFGLZM64LDAVCNFSM6AAAAABIXKU6BOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCOBVG42TOMZQGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Pedro-Roque commented:
Done. Will wait for @ahcorde and @azeey for inputs

Crola1702 commented:
I was trying to bring up and agent and I think gz_sim-ci-pr_any-focal-amd64  failure was my fault, sorry

@osrf-jenkins retest this please

arjo129 commented:
Can you merge with gz-sim7 to make the ABI checker happy

Pedro-Roque commented:
@arjo129 just merged it in, thanks for noting that!

bperseghetti commented:
@Pedro-Roque Thanks for your contribution, note this will not be in the binaries till a release is cut and run (including all other needed backports/forwardports that the team thinks is needed to allow for a release. I will see about the timing of that this coming Monday.

Pedro-Roque commented:
Thanks @bperseghetti - how can I get that info on Monday? Reach out via Discord?

bperseghetti commented:
> Thanks @bperseghetti - how can I get that info on Monday? Reach out via Discord?

I'll ping you on here.

Pedro-Roque commented:
@bperseghetti just pinging for further info. A release with these additions would strongly simplify our setup

bperseghetti commented:
> @bperseghetti just pinging for further info. A release with these additions would strongly simplify our setup

You can follow it here: https://github.com/gazebosim/gz-sim/pull/2491

Pedro-Roque commented:
@bperseghetti just checking if there's any reason why the plugin was not propagated to newer gazebo versions - shall I do a PR for them? It is missing in harmonic, at least.

iche033 commented:
forward ported to harmonic in https://github.com/gazebosim/gz-sim/pull/2504. I think there just needs to be a new gz-sim8 release.

bperseghetti commented:
> forward ported to harmonic in #2504. I think there just needs to be a new gz-sim8 release.

@Pedro-Roque I can open a release request PR this evening.

Pedro-Roque commented:
Thanks @bperseghetti and @iche033 ! When are the binaries going to be released for Ubuntu 24?

bperseghetti commented:
> Thanks @bperseghetti and @iche033 ! When are the binaries going to be released for Ubuntu 24?

Should be covered by this: https://github.com/gazebosim/gz-sim/pull/2671
